lib/cyclonedx/bom_builder.rb

```diff
@@ -38,34 +59,62 @@ def self.build(path)
           puts "#{@gems.size} gems were written to BOM located at #{@bom_file_path}"
         end
       rescue StandardError => e
-        @logger.error("Unable to write BOM to #{@bom_file_path}. #{e.message}: #{e.backtrace.join('\n')}")
+        @logger.error("Unable to write BOM to #{@bom_file_path}. #{e.message}: #{Array(e.backtrace).join("\n")}")
         abort
       end
+
+      if @options[:validate]
+        success, message = validate_bom_content(bom, @bom_output_format, @spec_version)
+        unless success
+          @logger.error(message)
+          exit(1)
+        end
+        @logger.info("BOM validation succeeded for spec #{@spec_version}") if @options[:verbose]
+      end
     end
 
-    def self.setup(_path)
+    # Infer format from file extension when not explicitly provided
+    def self.infer_format_from_path(path)
+      File.extname(path).downcase == '.json' ? 'json' : 'xml'
+    end
+
+    private
+
+    def self.setup(path)
       @options = {}
       OptionParser.new do |opts|
         opts.banner = 'Usage: cyclonedx-ruby [options]'
 
         opts.on('-v', '--[no-]verbose', 'Run verbosely') do |v|
           @options[:verbose] = v
         end
-        opts.on('-p', '--path path', '(Required) Path to Ruby project directory') do |path|
-          @options[:path] = path
+        opts.on('-p', '--path path', '(Required) Path to Ruby project directory') do |proj_path_opt|
+          @options[:path] = proj_path_opt
         end
         opts.on('-o', '--output bom_file_path', '(Optional) Path to output the bom.xml file to') do |bom_file_path|
           @options[:bom_file_path] = bom_file_path
         end
         opts.on('-f', '--format bom_output_format', '(Optional) Output format for bom. Currently support xml (default) and json.') do |bom_output_format|
           @options[:bom_output_format] = bom_output_format
         end
+        opts.on('-s', '--spec-version version', '(Optional) CycloneDX spec version to target (default: 1.7). Supported: 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7') do |spec_version|
+          @options[:spec_version] = spec_version
+        end
+        opts.on('--validate', 'Validate the produced BOM against the selected CycloneDX schema') do
```

Comment:
I find these two options to be a bit weird, if the produced BOM did not validate I would see that as a bug.